Retire `phpcs-diff`

eason9487 opened the pull request:
### Changes proposed in this Pull Request:

This PR retires `phpcs-diff` and add the `phpcs` workflow.

### Checks:

* [x] Does your code follow the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Detailed test instructions:

1. `npm run lint:php` to see if the check can pass
2. View [the check result](https://github.com/woocommerce/woocommerce-google-analytics-integration/actions/runs/9710643284/job/26801748611) of the `phpcs` workflow
3. View [the annotations of linting errors](https://github.com/woocommerce/woocommerce-google-analytics-integration/pull/446/files/0e2ce9b5ef98549a83d8d1dd581622ddfc790211) added by the `phpcs` workflow
   ![image](https://github.com/woocommerce/woocommerce-google-analytics-integration/assets/17420811/89a562fa-414a-4b64-b975-d2fd788ca68a)

### Changelog entry

eason9487 committed:
Replace the `phpcs-diff` workflow with the `phpcs` one.

eason9487 committed:
Remove unneeded package and scripts.

eason9487 committed:
Use `phpcs:ignore` instead of `phpcs:disable` to skip single-line checks.

eason9487 committed:
Remove `phpcs:ignoreFile` flag from the tests/class-unittestsbootstrap.php

eason9487 committed:
Skip ignorable phpcs checks in the tests/class-unittestsbootstrap.php

puntope commented on tests/class-unittestsbootstrap.php:
```diff
@@ -69,8 +70,10 @@ public function set_path_props() {
 	 * Set server props
 	 */
 	public function set_server_props() {
+		// phpcs:disable WordPress.Security.ValidatedSanitizedInput.MissingUnslash, WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
```
Do you think we can sanitize the input here instead of ignoring the line?

eason9487 replied:
Since this file is used for PHPUnit testing, I thought it would be okay to skip the checks.

I revisited its logic once and found that after the rewrite it didn't have to skip checking. Adjusted in https://github.com/woocommerce/woocommerce-google-analytics-integration/pull/446/commits/db107f7d216adbd9798695e9c8bd05ddce288a67.

puntope reviewed (approved):
Thanks @eason9487 for opening this PR

I tested it and works

<img width="836" alt="Screenshot 2024-06-28 at 18 16 25" src="https://github.com/woocommerce/woocommerce-google-analytics-integration/assets/5908855/172350cd-ceac-4d1d-ac47-659f03ff7bb8">


I left one comment regarding the handle of the PHPCS exceptions instead of ignoring them if possible, in this PR or maybe create an Issue for it.

Approved tho since it's not a blocker because it was not introduced in this PR.

eason9487 committed:
Rewrite `UnitTestsBootstrap::set_server_props` to avoid needing to skip phpcs checks.

Address: https://github.com/woocommerce/woocommerce-google-analytics-integration/pull/446#discussion_r1658827528
